app/Livewire/Concerns/Followable.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Livewire\Concerns;
+
+use App\Models\User;
+use Illuminate\Support\Number;
+use Livewire\Attributes\Renderless;
+
+trait Followable
+{
+    /**
+     * Follows the given user.
+     */
+    #[Renderless]
+    public function follow(int $id): void
+    {
+        if (! auth()->check()) {
+            $this->redirectRoute('login', navigate: true);
+
+            return;
+        }
+
+        $user = type(auth()->user())->as(User::class);
+        $user->following()->attach($id);
+
+        if ($this->shouldHandleFollowingCount()) {
+            $followingCount = $user->following()->count();
+            $this->dispatch('following.updated',
+                followingCount: $followingCount,
+                abbrFollowingCount: Number::abbreviate($followingCount),
+            );
+        }
+
+        $this->dispatch('user.followed', id: $id);
+    }
+
+    /**
+     * Unfollows the given user.
+     */
+    #[Renderless]
+    public function unfollow(int $id): void
+    {
+        if (! auth()->check()) {
+            $this->redirectRoute('login', navigate: true);
+
+            return;
+        }
+
+        $user = type(auth()->user())->as(User::class);
+        $user->following()->detach($id);
+
+        if ($this->shouldHandleFollowingCount()) {
+            $followingCount = $user->following()->count();
+            $this->dispatch('following.updated',
```

Comment:
can we avoid this, and just refresh the entire component on the front end?

Reply:
done